release/v1alpha1/types.go

```diff
@@ -0,0 +1,171 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ReleasePayload encapsulates the information for the creation of the ReleasePayload
+// and aggregates the outputs of the verification tests.
+// The release-controller monitors a specific set of imagestreams for updates.  If/when
+// an update occurs to these imagestreams, the release-controller will:
+//   1) Create a new ReleasePayload
+//      a: in the namespace configured by the --release-namespace parameter of the release-controller
+//      b. named after the name or mirrorPrefix in the aforementioned monitored imagestreams
+//   2) Start any number of tests specified as an annotation on the aforementioned monitored imagestreams
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type ReleasePayload struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec the inputs used to create the ReleasePayload
+	Spec ReleasePayloadSpec `json:"spec,omitempty"`
+
+	// Status is the current status of the ReleasePayload
+	Status ReleasePayloadStatus `json:"status,omitempty"`
+}
+
+// ReleasePayloadSpec has the information to represent a PromotionTest
+type ReleasePayloadSpec struct {
+	// Source the name of the imagestream where this imagestream (aka mirror) was created from
```

Comment:
imagestreams are namespaced.  Is the releasepayload in the same namespace as the imagestream?